File: Pc191/LeetCode/No_consecutive_1.cpp
#include <bits/stdc++.h>
using namespace std;

vector<string> sol(int n)
{
    vector<string> result;
    string path = "";

    function<void(int)> backtrack = [&](int index)
    {
        if (n == index)
        {
            result.push_back(path);
            return;
        }

        path.push_back('0');
        backtrack(index + 1);
        path.pop_back();

        if (path.empty() || path.back() == '0')
        {
            path.push_back('1');
            backtrack(index + 1);
            path.pop_back();
        }
    };

    backtrack(0);
    return result;
}

int main()
{
    vector<string> ans = sol(3);

    for (string s : ans)
    {
        cout << s << " ";
    }
}

File: Pc191/LeetCode/no_consecutive_1.py
def sol(n:int)->list[str]:
    result=[]
    path=[]
    
    def backtrack(index:int):
        if n==index:
            result.append("".join(path))
            return
        
        path.append('0')
        backtrack(index+1)
        path.pop()
        
        if len(path)==0 or path[-1]=='0':
            path.append('1')
            backtrack(index+1)
            path.pop()
            
    
    backtrack(0)
    return result


